components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -57,9 +151,187 @@ bool BravePrefProvider::SetWebsiteSetting(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+      brave_cookie_rules_[incognito].push_back(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+    }
+  }
+
+  // get the list of changes
+  std::vector<Rule> brave_cookie_updates;
+  for (const auto& new_rule : brave_cookie_rules_[incognito]) {
+    auto match = std::find_if(
+        old_rules.begin(),
+        old_rules.end(),
+        [&new_rule](const auto& old_rule) {
+          // we want an exact match here because any change to the rule
+          // is an update
+          return new_rule.primary_pattern == old_rule.primary_pattern &&
+                 new_rule.secondary_pattern == old_rule.secondary_pattern &&
+                 ValueToContentSetting(&new_rule.value) ==
+                    ValueToContentSetting(&old_rule.value);
+        });
+    if (match == old_rules.end()) {
+      brave_cookie_updates.push_back(CloneRule(new_rule));
+    }
+  }
+
+  // find any removed rules
+  for (const auto& old_rule : old_rules) {
+    auto match = std::find_if(
+        brave_cookie_rules_[incognito].begin(),
+        brave_cookie_rules_[incognito].end(),
+        [&old_rule](const auto& new_rule) {
+          // we only care about the patterns here because we're looking
+          // for deleted rules, not changed rules
+          return new_rule.primary_pattern == old_rule.primary_pattern &&
+                 new_rule.secondary_pattern == old_rule.secondary_pattern;
+        });
+    if (match == brave_cookie_rules_[incognito].end()) {
+      brave_cookie_updates.push_back(
+          Rule(old_rule.primary_pattern,
+               old_rule.secondary_pattern,
+               base::Value()));
+    }
+  }
+
+  // Notify brave cookie changes as CONTENT_SETTINGS_TYPE_COOKIES
+  if (content_type == CONTENT_SETTINGS_TYPE_PLUGINS) {
+    // PostTask here to avoid content settings autolock DCHECK
+    base::PostTaskWithTraits(
+          FROM_HERE,
+          {content::BrowserThread::UI, base::TaskPriority::USER_VISIBLE},
+          base::BindOnce(&BravePrefProvider::NotifyChanges,
+                         weak_factory_.GetWeakPtr(),
+                         std::move(brave_cookie_updates),
+                         incognito));
+  }
+}
+
+void BravePrefProvider::NotifyChanges(const std::vector<Rule>& rules,
+                                      bool incognito) {
+  for (const auto& rule : rules) {
+    Notify(rule.primary_pattern,
+           rule.secondary_pattern,
+           CONTENT_SETTINGS_TYPE_COOKIES,
+           "");
+  }
+}
+void BravePrefProvider::OnCookieSettingsChanged(ContentSettingsType content_type) {
+  UpdateCookieRules(content_type, true);
+  UpdateCookieRules(content_type, false);
+}
+
+void BravePrefProvider::OnContentSettingChanged(
```

Comment:
Just to improve my understanding - what triggers it?

Reply:
`ObservableProvider::NotifyObservers` - we register as an observer for that here https://github.com/brave/brave-core/pull/3130/files#diff-9981585c3d9184fc54b96697c59d2028R127